### ISSUE TITLE
v2.0

### ISSUE THREAD

**Issue by murilobom** [jhuckaby/webcamjs#242]
When will you release version 2.0?

**Comment by jhuckaby** [comment]
I apologize for the delay on v2.0.  I've been working on an image manipulation library for adding effects to the Webcam, and a much better way to deal with cropping / resizing, and output options (8-bit PNG, GIF, with quantization and diffusion dither, etc.).  The image library should be done in a couple weeks, then I'll return to working on WCJS 2.0 core.

My latest estimate is about a month, I think.

**Comment by prcongithub** [comment]
Any update here? I love this library and can't wait to see its 2.0.

**Comment by danielecacone** [comment]
I love this library too, I'm looging forwart trying the next realise :-)

**Comment by ScottDellinger** [comment]
danielcacone: You can already use the rear camera on mobile phones. The browser is the interface, though... not webcam.js

**Comment by maheshvartak3** [comment]
The mobile browser takes images from front camera be default. It would be better if there is setting for camera selection OR set default rear camera specially on mobile devices.

**Comment by ScottDellinger** [comment]
Any word on v2?

**Comment by jhuckaby** [comment]
Ah, so sorry everyone, a million things happened in real life, and I had to set aside this project for a while.  I did finish the image manipulation library (for adding effects, etc.), but need to write docs for it, then get back to working on WCJS v2.  Also, the Adobe news that Flash was being discontinued made me think this library may not really be needed much anymore, so it hasn't really been high priority.

I haven't given up tho!  I will keep trudging forward on v2.0.

**Comment by soryy708** [comment]
Flash being discontinued isn't a death blow to the library.
It's a nice fallback option to have.

The strength of the library is how many fallbacks it has and how widely supported it is.
If only Safari support was a little better... That's the main thing I'm looking forward to.

**Comment by Roldification** [comment]
Hi I am just here commenting in 2019 patiently waiting for the v.2 release. I am earnestly hoping for the author to keep this up because this really is a breakthrough in web development and I really appreciate and grateful to him for this.

**Comment by jhuckaby** [comment]
Hi there!  Author here.  I am sorry I have been absent for so long, and not responding to issues on this repo.  The death of Flash really took the wind out of my sails on v2.  Truly, the original and sole purpose of WebcamJS was to utilize Flash to take snapshots (because it was the only way back then).  But now Flash is very dead, and the HTML5 [MediaDevices](https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices) API has taken over.

Is v2 still needed?  I guess what I am saying here is, now that the MediaDevices API is widely supported across all browsers, would a simple wrapper like WebcamJS still be valuable to you?

**Comment by ssilverm** [comment]
Yes.  Yes it would.  Simple wrappers make my life so much easier!

**Comment by lofcz** [comment]
Alas it would be a waste not to share what you've created so far for v2. Ship it! <3

**Comment by overcatbe** [comment]
Hi, still waiting for version 2.0
I have a problem with newest apple devices.
Could you pls update?

**Comment by ErpMstar** [comment]
How to switch camera, When we have multiple cameras? I mean how to enable the camera selector option.

**Comment by niente0** [comment]
Hello, are you still working on v2? Thank you
